static/js/main.js

```diff
@@ -62,6 +62,18 @@ var header = {
         var password = $("#password").val();
         var verify_password = $("#verify_password").val();
 
+        if ($("#user_id")) {
+            var reg = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+            if (reg.test(user_id) == false) {
+                $.notify({
+                message: 'Please enter a valid email address!'
```

Comment:
Will it be possible to elaborate this notification to help the user know what is wrong with the email address?

Reply:
We can do that if split the regular expression case by case. I am going to let @lohani2280  document the regular expression above and then we can see what we should notify a user about.